packages/flutter_tools/lib/src/test/flutter_web_platform.dart

```diff
@@ -702,3 +764,179 @@ class _BrowserEnvironment implements Environment {
   @override
   CancelableOperation<dynamic> displayPause() => _manager._displayPause();
 }
+
+/// Helper class to start golden file comparison in a separate process.
+///
+/// Golden file comparator is configured using flutter_test_config.dart and that
+/// file can contain arbitrary Dart code that depends on dart:ui. Thus it has to
+/// be executed in a `flutter_tester` environment. This helper class generates a
+/// Dart file configured with flutter_test_config.dart to perform the comparison
+/// of golden files.
+class _TestGoldenComparator {
```

Comment:
This is a really interesting approach. i think this PR should have at least one end to end test that goes through this path, if possible.

Reply:
Added tests for `TestGoldenComparator` and `TestGoldenComparatorProcess`, PTAL, thanks.